### PR TITLE
Move /gas, /check-verification-status, /email-signup, /luma-events APIs to edge

### PR DESCRIPTION
**Pull request by MananTank**
`@thirdweb-dev/sdk` has tree-shaking issue and that bundle is not edge-runtime friendly - so edge functions must not contain any imports of `@thirdweb-dev/sdk`

### CODE REVIEW

**Comment by jnsdls** [comment]
#### 📝 Changed routes:



- [**/api/check-verification-status**](https://thirdweb-lqoyay8o3.thirdweb-preview.com/api/check-verification-status)
- [**/api/email-signup**](https://thirdweb-lqoyay8o3.thirdweb-preview.com/api/email-signup)
- [**/api/etherscan-fetch**](https://thirdweb-lqoyay8o3.thirdweb-preview.com/api/etherscan-fetch)
- [**/api/gas**](https://thirdweb-lqoyay8o3.thirdweb-preview.com/api/gas)
- [**/api/verify**](https://thirdweb-lqoyay8o3.thirdweb-preview.com/api/verify)

1 deleted route:
- **/api/paper/register**

Commit f0991c186d02466e8e2bd0466c49222dafe7a528 (https://thirdweb-lqoyay8o3.thirdweb-preview.com).

**Comment by jnsdls** [comment]
> 

Just another reason to get that fixed :)

**Comment by jnsdls** [comment]
> `@thirdweb-dev/sdk` has tree-shaking issue and that bundle is not edge-runtime friendly - so edge functions must not contain any imports of `@thirdweb-dev/sdk`

Just another reason to get that fixed :)

**Comment by jnsdls** [comment]
looks good @MananTank what do we need to test to be able to get this out?

File: pages/api/check-verification-status.ts [hunk 1]
```diff
@@ -19,12 +24,12 @@ const handler = async (req: NextApiRequest, res: NextApiResponse) => {
 
     const data = await result.json();
     if (data.status === "1") {
-      return res.status(200).json({ result: data.result });
+      return NextResponse.json({ result: data.result }, { status: 200 });
```
Comment (jnsdls) [inline pages/api/check-verification-status.ts:27]: in this case we should add a cache flag here (we want the client to cache this response forever really, a contract can never be un-verified)

File: pages/api/check-verification-status.ts [hunk 2]
```diff
@@ -19,12 +24,12 @@ const handler = async (req: NextApiRequest, res: NextApiResponse) => {
...
     } else {
-      return res.status(200).json({ result: data.result });
+      return NextResponse.json({ result: data.result }, { status: 200 });
```
Comment (jnsdls) [inline pages/api/check-verification-status.ts:29]: no cache in the case that it's not verified (user may verify immediately after and we don't want to hit cache in that case)

**Review by jnsdls — COMMENTED** [review]
@MananTank 1 small nitpick that we can improve later, for now we :shipit:

File: pages/api/check-verification-status.ts (hunk 1 above)
Comment (MananTank) [reply pages/api/check-verification-status.ts:27]: Done

**Pull request merged**